operator/hack/build-index-image.sh

```diff
@@ -0,0 +1,145 @@
+  --clean-output-dir   Delete '{base-dir}/build/index' directory.
+  --use-http           Use plain HTTP for container image registries.
+" >&2
+}
+
+function usage_exit() {
+  usage
+  exit 1
+}
+
+# Script argument variables
+BASE_INDEX_TAG=""
+INDEX_TAG=""
+BUNDLE_TAG=""
+REPLACED_VERSION=""
+BASE_DIR="."
+
+# Helpful for local development and testing
+CLEAN_OUTPUT_DIR=""
+USE_HTTP=""
+
+function read_arguments() {
+    while [[ -n "${1:-}" ]]; do
+        case "${1}" in
+            "--base-index-tag")
+                BASE_INDEX_TAG="${2}";shift;;
+            "--index-tag")
+                INDEX_TAG="${2}";shift;;
+            "--bundle-tag")
+                BUNDLE_TAG="${2}";shift;;
+            "--replaced-version")
+                REPLACED_VERSION="${2}";shift;;
+            "--base-dir")
+                BASE_DIR="${2}";shift;;
+            "--clean-output-dir")
+                CLEAN_OUTPUT_DIR="true";;
+            "--use-http")
+                USE_HTTP="--use-http";;
+            *)
+                echo "Error: Unknown parameter: ${1}" >&2
+                usage_exit
+        esac
+
+        if ! shift; then
+            echo 'Error: Missing parameter argument.' >&2
+            usage_exit
+        fi
+    done
+}
+
+function validate_arguments() {
+  [[ "${BASE_INDEX_TAG}" = "" ]] && echo "Error: Base index tag is required." >&2 && usage_exit
+  [[ "${INDEX_TAG}" = "" ]] && echo "Error: Index tag is required." >&2 && usage_exit
+  [[ "${BUNDLE_TAG}" = "" ]] && echo "Error: Bundle tag is required." >&2 && usage_exit
+  [[ "${REPLACED_VERSION}" = "" ]] && echo "Error: Replaced version is required." >&2 && usage_exit
+
+  return 0
+}
+
+SCRIPT_DIR=$(dirname "${BASH_SOURCE[0]}")
+
+OPM="opm"
+function fetch_opm() {
+  local -r os_name=$(uname | tr '[:upper:]' '[:lower:]') || true
+  local -r arch=$(go env GOARCH) || true
+
+  OPM="${BASE_DIR}/bin/opm-${OPM_VERSION}"
+  "${SCRIPT_DIR}/get-github-release.sh" --to "${OPM}" --from "https://github.com/operator-framework/operator-registry/releases/download/v${OPM_VERSION}/${os_name}-${arch}-opm"
+}
+
+YQ="yq"
+function fetch_yq() {
+  local -r os_name=$(uname | tr '[:upper:]' '[:lower:]') || true
+  local -r arch=$(go env GOARCH) || true
+
+  YQ="${BASE_DIR}/bin/yq-${YQ_VERSION}"
+  "${SCRIPT_DIR}/get-github-release.sh" --to "${YQ}" --from "https://github.com/mikefarah/yq/releases/download/v${YQ_VERSION}/yq_${os_name}_${arch}"
+}
+
+# Script body
+read_arguments "$@"
+validate_arguments
+fetch_opm
+fetch_yq
+
+if [[ "${CLEAN_OUTPUT_DIR}" = "true" ]]; then
+  rm -rf "${BASE_DIR}/build/index"
+fi
+
+if [[ "${REPLACED_VERSION}" = v* ]]; then
+  REPLACED_VERSION="${REPLACED_VERSION:1}"
+fi
+
+echo "Detected that bundle ${BUNDLE_TAG} updates (replaces) version ${REPLACED_VERSION}. Will use index image ${BASE_INDEX_TAG} as the base for the current one." >&2
+
+# Exports for docker build and opm in case it builds the image
+export DOCKER_BUILDKIT=1
+export BUILDKIT_PROGRESS="plain"
+
```

Comment:
As I understand, it's simply a flag for Docker to use BuildKit. It has a better performance than the default Docker build. So, I would leave it because it will keep our CI pipelines faster.